File: solid-principles/java/java-dip-notification/Mail.java
public class Mail implements Communication {
    public void sendInfo(User user, String message){
        this.sendMail(user.getMail(), message);
    }

    public void sendMail(String mail, String message){
        System.out.println("The message (" + message + ") has been sent to the " + mail + " email address.");
    }
}

File: solid-principles/python/python-dip-notification/Mail.py
from abc import ABC , abstractmethod
from Communication import Communication
from User import User

class Mail(Communication):
    def send_info(self, user, message):
        self.send_mail(user.get_mail(), message)
    def send_mail(self, mail, message):
        print(f"The message ({message}) has been sent to the mail {mail} email address.")    

        